components/dex.libsonnet

```diff
@@ -1,28 +1,90 @@
-{
+// These are the defaults for this components configuration.
+// When calling the function to generate the component's manifest,
+// you can pass an object structured like the default to overwrite default values.
+local defaults = {
+  local defaults = self,
+  name: error 'must provide name',
+  image: error 'must provide image',
+  version: error 'must provide version',
+  namespace: error 'must provide namespace',
+  config: {
+    issuer: 'http://%s.%s.svc.cluster.local:5556/dex' % [defaults.name, defaults.namespace],
+    storage: {
+      type: 'sqlite3',
+      config: { file: '/storage/dex.db' },
+    },
+    web: { http: '0.0.0.0:5556' },
+    logger: { level: 'debug' },
+  },
+  ports: { http: 5556 },
+
+  commonLabels:: {
+    'app.kubernetes.io/name': 'dex',
+    'app.kubernetes.io/instance': defaults.name,
+    'app.kubernetes.io/version': defaults.version,
+    'app.kubernetes.io/component': 'identity-provider',
+  },
+};
+
+function(params) {
   local dex = self,
 
-  config:: {
-    name: error 'must provide name',
-    image: error 'must provide image',
-    version: error 'must provide version',
-    namespace: error 'must provide namespace',
-    config: {
-      issuer: 'http://%s.%s.svc.cluster.local:5556/dex' % [dex.config.name, dex.config.namespace],
-      storage: {
-        type: 'sqlite3',
-        config: { file: '/storage/dex.db' },
+  // Combine the defaults and the passed params to make the component's config.
+  config:: defaults + params,
+
+  secret: {
+    apiVersion: 'v1',
+    stringData: {
+      'config.yaml': std.manifestYamlDoc(dex.config.config),
+    },
+    kind: 'Secret',
+    metadata: {
+      labels: dex.config.commonLabels,
+      name: dex.config.name,
+      namespace: dex.config.namespace,
+    },
+  },
+
+  pvc: {
```

Comment:
in kube-thanos we used volumeClaimTemplate here. It's fine probably cause it was called `pvc` before, just slight inconsistencies, I guess.

Reply:
I'm gonna address this one before we merge